src/pmacfiltercontrol/pmacFilterControlWrapper.py

```diff
@@ -491,6 +491,13 @@ def _set_mode(self, mode: int) -> None:
 
         self.mode_rbv.set(mode)
 
+        if self.h5f is not None:
+            try:
+                self._close_file()
+            except Exception as e:
+                print(f"Failed closing file.\n{e}")
+
+
```

Comment:
I would extract this into a method as you do exactly the same thing [here](https://github.com/dls-controls/pmacFilterControl/pull/43/files#diff-d20b1c35e003f54bbf161323660dea90d877069640a1401145ad7a14d212997aL470)